src/Microsoft.Framework.Logging/Utils/TypeHelper.cs

```diff
@@ -0,0 +1,112 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Reflection;
+using System.Text;
+
+namespace Microsoft.Framework.Logging
+{
+    public static class TypeNameHelper
+    {
+        public static string GetTypeDisplayFullName(Type type)
+        {
+            return GetSimpleTypeName(type);
+        }
+
+        private static string GetSimpleTypeName(Type t)
+        {
+            StringBuilder sb = new StringBuilder(64);
+            ProcessTypeName(t, delegate (TypeWrapper writerItem)
```

Comment:
I think this should be simplified to not use the delegate (or lambda). Can just have a helper method that does this work and is called from wherever this delegate is otherwise passed around. I think that'll make the code quite simpler (just pass the StringBuilder around instead).